agent/api/task.go

```diff
@@ -66,32 +66,39 @@ type Task struct {
-	// backend as a result of either API calls made to ECS or decisions made by the ECS service scheduler.  The
-	// DesiredStatus is almost always either TaskRunning or TaskStopped.  Do not access DesiredStatus directly.  Instead,
-	// use `UpdateStatus`, `UpdateDesiredStatus`, `SetDesiredStatus`, and `SetDesiredStatus`.
-	// TODO DesiredStatus should probably be private with appropriately written setter/getter.  When this is done, we need
-	// to ensure that the UnmarshalJSON is handled properly so that the state storage continues to work.
+	// DesiredStatus represents the state where the task should go. Generally,
+	// the desired status is informed by the ECS backend as a result of either
+	// API calls made to ECS or decisions made by the ECS service scheduler.
+	// The DesiredStatus is almost always either TaskRunning or TaskStopped.
+	// NOTE: Do not access DesiredStatus directly.  Instead, use `UpdateStatus`,
+	// `UpdateDesiredStatus`, `SetDesiredStatus`, and `SetDesiredStatus`.
+	// TODO DesiredStatus should probably be private with appropriately written
+	// setter/getter.  When this is done, we need to ensure that the UnmarshalJSON
+	// is handled properly so that the state storage continues to work.
 	DesiredStatus     TaskStatus
 	desiredStatusLock sync.RWMutex
 
-	// KnownStatus represents the state where the task is.  This is generally the minimum of equivalent status types for
-	// the containers in the task; if one container is at ContainerRunning and another is at ContainerPulled, the task
-	// KnownStatus would be TaskPulled.  Do not access KnownStatus directly.  Instead, use `UpdateStatus`,
-	// `UpdateKnownStatusAndTime`,  and `GetKnownStatus`.
-	// TODO KnownStatus should probably be private with appropriately written setter/getter.  When this is done, we need
-	// to ensure that the UnmarshalJSON is handled properly so that the state storage continues to work.
+	// KnownStatus represents the state where the task is.  This is generally
+	// the minimum of equivalent status types for the containers in the task;
+	// if one container is at ContainerRunning and another is at ContainerPulled,
+	// the task KnownStatus would be TaskPulled.
+	// NOTE: Do not access KnownStatus directly.  Instead, use `UpdateStatus`,
+	// and `GetKnownStatus`.
+	// TODO KnownStatus should probably be private with appropriately written
+	// setter/getter.  When this is done, we need to ensure that the UnmarshalJSON
+	// is handled properly so that the state storage continues to work.
 	KnownStatus     TaskStatus
 	knownStatusLock sync.RWMutex
-	// KnownStatusTime captures the time when the KnownStatus was last updated.  Do not access KnownStatusTime directly,
-	// instead use `GetKnownStatusTime`.
+	// KnownStatusTime captures the time when the KnownStatus was last updated.
+	// NOTE: Do not access KnownStatusTime directly, instead use `GetKnownStatusTime`.
```

Comment:
Though thinking about this more, we could rename `KnownStatusTime` (and others) to `KnownStatusTimeUnsafe` and still let it marshal with the `json:"KnownTime"` tag.